doc/style-guide-testing.md

```diff
@@ -0,0 +1,467 @@
+Writing High Quality Tests
+==========================
+
+The goal of a test is to ensure that some bit of code works as expected with valid or invalid input.  Testing failure paths is as important as testing successful paths.
+
+Tests also provide a way of proving that your interface doesn't change.  People will write code that use your module in a defined way, which should mirror a test.  When you refactor code for efficiency, to split into smaller modules or to use a different method for performing the same action your tests will guarantee that everything still works.
+
+Think of tests as a safety net for acrobats.  You'd rather be perfect and perform your art without ever needing tests, but they are there to help you when a mistake is made.
+
+
+Guidelines
+----------
+
+The [style guide](style-guide.md) lists many useful tips and various requirements when writing tests.  Only a few of those are listed here for further explanation.
+
+
+### Eliminate Duplicated Setup
+
+When the code has two separate actions that it performs then you want to have the tests split into two separate `it()` calls.  These would require the same setup to achieve an identical scenario.  Move that into a `beforeEach()`.  When you have a complicated piece of code that has numerous logical branches, again you will want to split the setup into multiple `beforeEach()` calls, one for each branch.
```

Comment:
Should we mention `beforeEach` is for Jasmine? What if they want to use a different testing framework.